src/sources/file.rs

```diff
@@ -1450,6 +1451,13 @@ mod tests {
         let dir = tempdir().unwrap();
         let config = file::FileConfig {
             include: vec![PathBuf::from("tests/data/gzipped.log")],
+            // TODO: remove this once files are fingerprinted after decompression
+            //
+            // Currently, this needs to be smaller than the total size of the compressed file
+            // because the fingerprinter tries to read until a newline, which it's not going to see
+            // in the compressed data, or this number of bytes. If it hits EOF before that, it
+            // can't return a fingerprint because the value would change once more data is written.
+            max_line_bytes: 100,
```

Comment:
This is because we check this value before the decompression, right? If we were checking it after - it would just work. I presume this is tracked by https://github.com/timberio/vector/issues/5085.

Reply:
Correct, that's what I tried to explain with the comment.